common/elasticsearch/config.go

```diff
@@ -33,12 +33,23 @@ import (
 // Config for connecting to ElasticSearch
 type (
 	Config struct {
-		URL     url.URL           `yaml:url`     //nolint:govet
-		Indices map[string]string `yaml:indices` //nolint:govet
+		URL      url.URL           `yaml:"url"` //nolint:govet
+		Username *string           `yaml:"username"`
+		Password *string           `yaml:"password"`
```

Comment:
Do we even want to support putting passwords to yaml files? A more secure way I think would be to only accept passwords programmatically. Now that we added server options, this can be done elegantly.

Reply:
at least our current solution allows password in yaml:
https://github.com/temporalio/temporal/blob/master/common/service/config/config.go#L224

one more thing is `accept passwords programmatically` requires the user to use our service like lib, while currently users can basically directly use our published docker.

Reply:
one more thing: the user / pass can be passed in from environment vars: https://github.com/temporalio/temporal/blob/master/docker/config_template.yaml#L32
this is how the secrets were used in my prev company

Reply:
That's true. Why `*string` and not `string` like on the other cases?

Reply:
that is true

@linvon could you plz change the `*string` to `string`?

Reply:
Sure, I've changed it